Automattic/pocket-casts-android
# Bump recycler view version

## Description
This bumps recycler view version to stable `v1.3.0.`

It also removes unused view holders and removes `DisposeOnViewTreeLifecycleDestroyed` and `ViewCompositionStrategy` described in the previous interoperability guidelines from `FolderViewHolder` as described in the [release notes](https://developer.android.com/jetpack/androidx/releases/recyclerview#version_130_3).

## Testing Instructions
- Smoke test folders list.
- Check for any view holders that might be using previous interoperability guidelines.

## Checklist
 
#### I have tested any UI changes...
